lib/fizzy/execute.cpp

```diff
@@ -324,6 +326,20 @@ inline DstT extend(SrcT in) noexcept
         return DstT{in};
 }
 
+template <typename SrcT, typename DstT>
+inline bool trunc(OperandStack& stack) noexcept
+{
+    using boundaries = trunc_boundaries<SrcT, DstT>;
+
+    const auto input = stack.top().as<SrcT>();
+    if (input > boundaries::lower && input < boundaries::upper)
+    {
+        stack.top() = static_cast<DstT>(input);
+        return true;
+    }
+    return false;
```

Comment:
After the optimisation I'd add `assert(std::isnan(input) || input == inf || input == -inf);` so that it matches the spec.

Reply:
Ok, added correct variants of these asserts.

Reply:
Don't want to stretch this too much, but can we also have the static_assert for inf vs. boundaries? Obviously we cannot make the nan-check compile time :)